src/NServiceBus.Core.Tests/AssemblyScanner/AssemblyScannerTests.cs

```diff
@@ -37,32 +41,226 @@ public void Non_system_assemblies_should_be_included()
             Assert.IsTrue(AssemblyScanner.ReferencesNServiceBus(combine));
         }
 
         [Test]
-        public void ReferencesNServiceBus_requires_binding_redirect()
+        public void ReferencesNServiceBus_returns_false_for_no_reference()
         {
-            var combine = Path.Combine(GetTestAssemblyDirectory(), "AssemblyWithRefToSN.dll");
-            Assert.IsTrue(AssemblyScanner.ReferencesNServiceBus(combine));
+            var combine = Path.Combine(GetTestAssemblyDirectory(), "dotNet.dll");
+            Assert.IsFalse(AssemblyScanner.ReferencesNServiceBus(combine));
         }
 
         [Test]
-        public void ReferencesNServiceBus_returns_true_for_direct_reference()
+        public void Assemblies_with_direct_reference_are_included()
         {
-            var combine = Path.Combine(GetTestAssemblyDirectory(), "AssemblyWithReference.dll");
-            Assert.IsTrue(AssemblyScanner.ReferencesNServiceBus(combine));
+            var busAssembly = new DynamicAssembly("NServiceBus.Core.dll");
+            var assemblyWithReference = new DynamicAssembly("AssemblyWithReference.dll", new[] { busAssembly });
+
+            using (var dynamicDirectory = new DynamicAssemblyDirectory(busAssembly, assemblyWithReference))
+            {
+                var scanner = new AssemblyScanner(dynamicDirectory);
+                var result = scanner.GetScannableAssemblies();
+
+                Assert.AreEqual(2, result.Assemblies.Count);
+                Assert.IsTrue(result.Assemblies.Contains(assemblyWithReference));
+                Assert.IsTrue(result.Assemblies.Contains(busAssembly));
+            }
         }
 
         [Test]
-        public void ReferencesNServiceBus_returns_false_for_no_reference()
+        public void Assemblies_with_no_reference_are_excluded()
         {
-            var combine = Path.Combine(GetTestAssemblyDirectory(), "dotNet.dll");
-            Assert.IsFalse(AssemblyScanner.ReferencesNServiceBus(combine));
+            var busAssembly = new DynamicAssembly("NServiceBus.Core.dll");
+            var assemblyWithReference = new DynamicAssembly("AssemblyWithReference.dll", new[] { busAssembly });
+            var assemblyWithoutReference = new DynamicAssembly("AssemblyWithoutReference.dll");
+
+            using (var dynamicDirectory = new DynamicAssemblyDirectory(busAssembly, assemblyWithReference, assemblyWithoutReference))
+            {
+                var scanner = new AssemblyScanner(dynamicDirectory);
+                var result = scanner.GetScannableAssemblies();
+
+                Assert.AreEqual(2, result.Assemblies.Count);
+                Assert.IsTrue(result.Assemblies.Contains(assemblyWithReference));
+                Assert.IsTrue(result.Assemblies.Contains(busAssembly));
+                Assert.IsFalse(result.Assemblies.Contains(assemblyWithoutReference));
+            }
+        }
+
+        [Test]
+        public void Assemblies_which_reference_older_nsb_version_are_included()
```

Comment:
Not sure how this test can work since it wants to write two files with the same name (NServiceBus.Core.dll) to the same directory.